front/src/app/welcome/validate/validate.component.ts

```diff
@@ -0,0 +1,83 @@
+import { takeUntil } from 'rxjs/operators';
+import { Component, HostBinding, OnDestroy, OnInit } from '@angular/core';
+import { Subject } from 'rxjs';
+import { ActivatedRoute, Router } from '@angular/router';
+import { MatSnackBar } from '@angular/material/snack-bar';
+
+import Map from 'ol/Map';
+import VectorSource from 'ol/source/Vector';
+
+import { GeoHelper } from '../../_helpers/geoHelper';
+import { GeoshopUtils } from '../../_helpers/GeoshopUtils';
+import { Order, OrderItem} from '../../_models/IOrder';
+import { ApiOrderService } from '../../_services/api-order.service';
+import { ConfigService } from '../../_services/config.service';
+import { MapService } from '../../_services/map.service';
+import Geometry from 'ol/geom/Geometry';
+
+
+@Component({
+  selector: 'gs2-validate',
+  templateUrl: './validate.component.html',
+  styleUrls: ['./validate.component.scss']
+})
+export class ValidateComponent implements OnInit, OnDestroy {
+
+  @HostBinding('class') class = 'main-container';
+
+  private onDestroy$ = new Subject<void>();
+  private token: string;
+  order: Order;
+  orderitem: OrderItem;
+  minimap: Map;
+  vectorSource: VectorSource<Geometry>;
+
+
+  constructor(
+    private apiOrderService: ApiOrderService,
+    private snackBar: MatSnackBar,
+    private route: ActivatedRoute,
+    private router: Router,
+    private configService: ConfigService,
+    private mapService: MapService) {
+    this.route.params
+      .pipe(takeUntil(this.onDestroy$))
+      .subscribe(params => this.token = params.token);
+  }
+
+  ngOnInit(): void {
+    this.apiOrderService.getOrderItemByToken(this.token)
+      .subscribe(iOrderItem => {
+        if (iOrderItem) {
+          console.log(iOrderItem)
```

Comment:
I do not remember if you keep these or not (there's another one on line 57)

Reply:
Thanks, I should get rid of that!